### ISSUE TITLE
Mermaid subgraph background too dark

### ISSUE THREAD

**Issue by catanzaromj** [quarto-dev/quarto-cli#9178]
### Bug description

As described in Issue https://github.com/quarto-dev/quarto-cli/issues/5416, the background of a mermaid subgraph is too dark when creating a revealjs presentation. It may extend to other formats, but that is my use case. 

I think I found a temporary fix and a possible solution. As mentioned [here](https://github.com/quarto-dev/quarto-cli/issues/5416#issuecomment-2015067277), the variable mermaid-fg-color--lightest in your CSS/SCSS controls the subgraph background. For example, to set it to white:

```
/*-- scss:defaults --*/
//other stuff
$mermaid-fg-color--lightest:#FFFFFF;
```

In case it helps anyone debug this, [this line](https://github.com/quarto-dev/quarto-cli/blob/88c5518462d6e511d452428aa3d1a518a5c64b67/src/resources/formats/html/mermaid/embed-mermaid.css#L58) sets the subgraph background color to mermaid-fg-color--lightest, since subgraphs are cluster elements. This defaults to $mermaid-fg-color--lightest: lighten($body-color, 20%); according to the [docs](https://quarto.org/docs/authoring/diagrams.html#customizing-mermaid). Should it default to a background color instead of a foreground color?

### Steps to reproduce

The code from [this post](https://github.com/quarto-dev/quarto-cli/issues/5416#issuecomment-1675728518) reproduces it for me:

````quarto
---
title: "test"
format: revealjs
---

```{mermaid}
flowchart LR
  subgraph A [Layer 1]
    direction LR
    id1[Affine Function] --> id2[ReLU]
  end

  subgraph B [Layer 2]
    direction LR
    id2 --> id3[Affine Function]
  end
  
  subgraph C [Loss Function]
    direction LR
    id3 --> id4[MSE]
  end
```
````
Then run `quarto preview test.qmd`.

### Expected behavior

I expect it to be light:

```mermaid
flowchart LR
  subgraph A [Layer 1]
    direction LR
    id1[Affine Function] --> id2[ReLU]
  end

  subgraph B [Layer 2]
    direction LR
    id2 --> id3[Affine Function]
  end
  
  subgraph C [Loss Function]
    direction LR
    id3 --> id4[MSE]
  end
```
Again, taken from [this post](https://github.com/quarto-dev/quarto-cli/issues/5416#issuecomment-1675814612).

### Actual behavior

It is very dark. When I run `quarto preview test.qmd`, I get
![image](https://github.com/quarto-dev/quarto-cli/assets/25447423/d0d27f65-a77c-4dee-9455-fa83a3034b1f)

### Your environment

- IDE: nvim 0.10.0 (but I'm running `preview` from the command line)
- OS: Ubuntu 22.04
- quarto v1.4.550, installed from conda forge

### Quarto check output

```bash 
$ quarto check
Quarto 1.4.550
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.58.3: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.550
      Path: 

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path:
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2 (Conda)
      Path: 
      Jupyter: 5.5.0
      Kernels: 

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)
```

**Comment by mcanouil** [comment]
FYI, you could set a Mermaid theme, see <https://quarto.org/docs/authoring/diagrams.html#using-mermaids-built-in-themes>.
Reveal.js does not use Bootstrap contrary to other Quarto HTML based formats and since Mermaid.js does make a lot of releases with many changes, it's no surprise that Quarto is bit behind in terms of theming compatibility.
This will be improved as much as possible, meantime the more reliable (in regards to Mermaid.js updates) would be to using Mermaid internal themes such as the `neutral` theme:

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/e4b62d0a-9247-4df6-8a40-42963a3d921e">

**Comment by cderv** [comment]
Some current workaround in the meantime share at 
* https://github.com/quarto-dev/quarto-cli/issues/9259

**Comment by rcannood** [comment]
I'm closing #5416 in favour of creating separate, more specific issues. This issue seems like a good place to start!

I just updated the Mermaid demos in quarto to Quarto 1.5.54 using the Mermaid demos for Mermaid 10.2.4, and I can confirm that the described issue still persists in Quarto 1.5.54.

---- 

Mermaid docs on subgraphs: https://mermaid.js.org/syntax/flowchart.html#subgraphs
 
![Screenshot from 2024-07-16 22-15-25](https://github.com/user-attachments/assets/8f3b9840-7a77-44e8-a992-008c86007b2f)

----

Output of the same example in Quarto 1.5.54: https://mermaid-demos-in-quarto.netlify.app/syntax/flowchart#subgraphs

![Screenshot from 2024-07-16 22-16-12](https://github.com/user-attachments/assets/cf9700f8-bfc8-4738-bc09-7593c9f844cd)

**Comment by rcannood** [comment]
It should be noted that the subgraph colours simply seem to be swapped between dark and light mode, because when I view the same diagram in dark mode, it is suddenly very bright.

[Mermaid docs](https://mermaid.js.org/syntax/flowchart.html#subgraphs) in dark mode:

![Screenshot from 2024-07-16 22-29-29](https://github.com/user-attachments/assets/1e51e271-91cc-46f7-949d-429221498f6f)

[Quarto render](https://mermaid-demos-in-quarto.netlify.app/syntax/flowchart#subgraphs) in dark mode:

![Screenshot from 2024-07-16 22-29-34](https://github.com/user-attachments/assets/64885926-c3e6-450c-9b26-bff6f6dc3f52)

**Comment by duaneking** [comment]
I just want to point out for people with poor vision or who are visually disabled or who are just tired after a long day of looking at multiple monitors that this is an accessibility nightmare, and it actively creates a lot of pain and suffering.

I'm currently dealing with the problem where for some strange reason mermaid's official editor is making all subgraphs this grey color that you cant see anything in, and its absolutely terrible on my eyes

![Image](https://github.com/user-attachments/assets/4b76edc3-a2a5-4fd3-b461-8f2aef9e0de2)

**Comment by cderv** [comment]
@duaneking right now, as a users, a workaround is to tweak the styles to improve this.

````markdown
---
format: 
  html: default
---

```{mermaid}
flowchart TB
    c1-->a2
    subgraph ide1 [one]
    a1-->a2
    end
    
    style c1 fill: lightgray
    style a1 fill: lightgray
    style a2 fill: lightgray
    style ide1 fill: lightyellow
```
````

![Image](https://github.com/user-attachments/assets/3aed879e-92d5-49fc-b9de-cf02e123f2a8)


@gordonwoodhull as you are working on dark / light, did you look at mermaid yet ? Putting this on your radar as you may have insight. From https://github.com/quarto-dev/quarto-cli/issues/9178#issuecomment-2231774985 there may be an issue with dark / light color being not considered correctly.

**Comment by cscheid** [comment]
I don't think we should try to fix mermaid's light/dark mode for our light/dark work in 1.7, unless it turns out to be very simple.

**Comment by ColinFay** [comment]
For anybody needing a quick hack to be sure all rect are filled with a specific color (note that this will apply to all rectangle), you can add: 

```
<style>
.mermaid-js rect{
  fill: lightyellow!important;
}
</style>
```

before your mermaid block

**Comment by OlmosDeAguilera** [comment]
This solved the issue for me without affecting document type:

````
---
mermaid:
    theme: neutral
---

# General overview
```{mermaid}
flowchart LR
    A[Hard edge] --> B(Round edge)
    B --> C{Decision}
    C --> D[Result one]
    C --> E[Result two]
```
```{mermaid}
flowchart LR

    subgraph Inputs
        A[Node A]
        B[Node B]
        C[Node C]
    end

    A --> D{Decision}
    B --> E[Result one]
    C --> F[Result two]
    
```
````
